### ISSUE TITLE
semantic: avoid resolving parameter initializers to later parameters

### ISSUE THREAD

**Issue by camc314** [oxc-project/oxc#22784]
## Summary

Oxc currently resolves an earlier parameter initializer to a parameter declared later in the same parameter list.

TypeScript treats this as invalid: a parameter initializer cannot reference itself or any parameter declared after it.

This is related to #22783, but it is a distinct path: the later parameter is already bound before Oxc performs its early parameter-reference resolution, so the reference resolves even though TypeScript would reject it.

## Repro

```ts
function f(a = b, b = 1) {
  return a;
}
```

TypeScript reports:

```text
TS2373: Parameter 'a' cannot reference identifier 'b' declared after it.
```

Oxc semantic currently resolves the `b` in `a = b` to the later parameter `b`.

Expected semantic behavior:

```text
a = b  // should not resolve to the later parameter b
```

## TypeScript behavior

TypeScript explicitly checks parameter initializer references and rejects references to later parameters:

https://github.com/microsoft/TypeScript/blob/f350b52331494b68c90ab02e2b6d0828d2a22a74/src/compiler/checker.ts#L3339-L3350

The broader lookup rules also restrict which function-local symbols are visible from parameter/signature positions:

https://github.com/microsoft/TypeScript/blob/f350b52331494b68c90ab02e2b6d0828d2a22a74/src/compiler/utilities.ts#L11545-L11581

## Oxc behavior

Oxc visits and binds all formal parameters, then resolves references from the parameter list:

https://github.com/oxc-project/oxc/blob/main/crates/oxc_semantic/src/builder.rs#L1925-L1949

Because all parameters have already been declared by the time resolution runs, an earlier initializer can resolve to a later parameter.

There is also an existing test that currently asserts this behavior:

https://github.com/oxc-project/oxc/blob/main/crates/oxc_semantic/tests/integration/scopes.rs#L135-L149

## Why this matters

This makes Oxc’s reference/declaration graph disagree with TypeScript for invalid parameter initializers. Tools consuming `oxc_semantic` may incorrectly treat a later parameter as referenced from an earlier initializer, even though TypeScript does not allow that binding.
